### ISSUE TITLE
Cloning objects that cannot be cloned throws exception

### ISSUE THREAD

**Issue by trajche** [digitalnodecom/node-red-contrib-generic-s3#64]
Since the latest update you use structuredClone and async processing. When cloning an object that contains a function it throws an exception that crashes node-red:

```
DataCloneError: function onReadableStreamEnd() {
  if (!this.allowHalfOpen) {
    this.write = writeAfterFIN;
  }
} could not be cloned.
```

If possible, can we have an error message rather than a crash stating an object cannot be cloned?

Maybe a check such as [this](https://stackoverflow.com/questions/32673518/how-to-check-if-an-object-can-be-cloned-by-the-structured-clone-algorithm)?

**Comment by rristov60** [comment]
Hi @trajche, thanks for opening this issue. This is being worked on and will be fixed fairly soon. It is planned for the node to throw an error which can be caught with `catch` node.

**Comment by rristov60** [comment]
@trajche this has been patched in the latest release `v1.12.2`
